src/components/BookList.js

```diff
@@ -1,29 +1,64 @@
-// components/BookList.js
-import React, { useState } from 'react';
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { addBook, removeBook } from '../redux/books/booksSlice';
 import Book from './Book';
 
 const BookList = () => {
-  const [books, setBooks] = useState([
-    { id: 1, title: 'Book 1', author: 'Author 1' },
-    { id: 2, title: 'Book 2', author: 'Author 2' },
-    // Add more books here
-  ]);
+  const books = useSelector((state) => state.books);
+  const dispatch = useDispatch();
 
   const handleDelete = (id) => {
-    setBooks(books.filter((book) => book.id !== id));
+    dispatch(removeBook({ id }));
+  };
+
+  const handleAddBook = () => {
+    const newBook = {
+      id: Date.now(),
+      title: 'New Book Title',
+      author: 'New Book Author',
+    };
+
+    dispatch(addBook(newBook));
+  };
+
+  const handleRemoveBook = () => {
+    if (books.length > 0) {
+      const lastBookId = books[books.length - 1].id;
+      dispatch(removeBook({ id: lastBookId }));
+    }
   };
 
   return (
-    <div>
-      <h1>Book List</h1>
-      {books.map((book) => (
-        <Book
-          key={book.id}
-          title={book.title}
-          author={book.author}
-          onDelete={() => handleDelete(book.id)}
-        />
-      ))}
+    <div className="container mx-auto px-4 py-8">
+      <h1 className="text-3xl font-bold mb-5 text-gray-800">Books</h1>
+      <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-4">
+        {books.map((book) => (
+          <Book
+            key={book.id}
+            title={book.title}
+            author={book.author}
+            onDelete={() => handleDelete(book.id)}
+          />
+        ))}
+      </div>
+      <div className="flex justify-center mt-5">
+        <button
+          type="button"
+          onClick={handleAddBook}
+          className="bg-green-500 text-white px-4 py-2 rounded-md mt-2 mr-2"
+        >
+          Add New Book
+        </button>
+        {books.length > 0 && (
+          <button
+            type="button"
+            onClick={handleRemoveBook}
+            className="bg-red-500 text-white px-4 py-2 rounded-md mt-2"
+          >
+            Remove Last Book
+          </button>
+        )}
+      </div>
```

Comment:
- Hi, can you make Adding and Removing the Book Possible? I am asking this because we need to check if the code you implemented is working accordingly. Then submit again for re-review if adding and removing is working.

![Capture](https://github.com/henask12/bookstore/assets/63972670/29b7694a-31f2-49d3-bf14-4b014476529b)